framework/src/task_manager/taskmgr_getinfo.c

```diff
@@ -0,0 +1,146 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <string.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <apps/builtin.h>
+#include <tinyara/taskmgt.h>
+#include <task_manager/task_manager.h>
+#include "task_manager_internal.h"
+
+/****************************************************************************
+ * private function
+ ****************************************************************************/
+static int tm_get_task_info(task_info_list_t **data, int handle)
+{
+	int len;
+	task_info_list_t *item;
+
+	if (IS_INVALID_HANDLE(handle)) {
+		return TM_INVALID_PARAM;
+	}
+
+	item = (task_info_list_t *)TM_ALLOC(sizeof(task_info_list_t));
+	if (item == NULL) {
+		tmdbg("Memory allocation Failed\n");
+		return TM_OUT_OF_MEMORY;
+	}
+
+	len = strlen(builtin_list[TASK_BUILTIN_IDX(handle)].name);
+	item->task.name = (char *)TM_ALLOC(len + 1);
+	if (item->task.name == NULL) {
+		tmdbg("Memory allocation for name Failed\n");
+		TM_FREE(item);
+		return TM_OUT_OF_MEMORY;
+	}
+	strncpy(item->task.name, builtin_list[TASK_BUILTIN_IDX(handle)].name, len + 1);
+	item->task.gid = TASK_TM_GID(handle);
+	item->task.handle = handle;
+	item->task.status = TASK_STATUS(handle);
+	item->task.permission = TASK_PERMISSION(handle);
+	item->next = *data;
+
+	*data = item;
+
+	return OK;
+}
+
+
+/****************************************************************************
+ * taskmgr_getinfo_with_name
```

Comment:
These functions have too less description. I don't know how to get information from function.

Reply:
Is it needed? these are internally used by task manager only.